packages/react-components/lib/tasks/task-timeline.tsx

```diff
@@ -9,51 +9,56 @@ import {
   TimelineItem,
   TimelineOppositeContent,
   TimelineSeparator,
-} from '@material-ui/lab';
+  TimelineProps,
+} from '@mui/lab';
 import React from 'react';
 import * as RmfModels from 'rmf-models';
 import { rosTimeToJs } from '../utils';
 
-const getPhaseColors = (theme: Theme) => ({
-  pending: theme.palette.info.light,
-  completed: theme.palette.success.light,
-  failed: theme.palette.error.light,
-});
+interface TimeLinePropsWithRef extends TimelineProps {
+  ref?: React.RefObject<HTMLUListElement>;
+}
```

Comment:
This type definition is potentially unsafe, we do not know if the ref is actually a `HTMLUListElement`, there are several mui components which you can customize to change the html tag, so there might not even be a fixed type.

Would it be possible to use `React.PropsWithRef`?

Reply:
It seems like the `TimeLine` RefObject type is fixed as `HTMLUListElement`. Can't use React.PropsWithRef